contrib/platform/test/com/sun/jna/platform/win32/Advapi32Test.java

```diff
@@ -1493,4 +1524,24 @@ private File createTempFile() throws Exception {
         fileWriter.close();
         return file;
     }
+    
+    public void testCreateProcessWithLogonW() {
+    	String winDir = Kernel32Util.getEnvironmentVariable("WINDIR");
+    	if (winDir == null || !(new File(winDir).exists())) {
+    		throw new IllegalStateException("WINDIR environment variable did not properly resolve to a directory.");
+    	}
```

Comment:
assertNotNull("No WINDIR value returned", winDir);
assertTrue("Specified WINDIR does not exist: " + winDir, new File(winDir).exists());